third_party/nvidia/hopper/lib/Transforms/WarpSpecialization/TMEMAlloc1D.cpp

```diff
@@ -93,15 +73,18 @@ class TMEM1DAllocator {
   }
 
   void TMEMStore1D(Operation *producer,
-                   std::optional<ttng::TMEMAllocOp> allocOpBuffer) {
+                   std::optional<Operation *> allocOpBuffer) {
     // Expand from 1D -> 2D
     auto oldRetType = getResultTensorType(producer, 1);
     builder.setInsertionPointAfter(producer);
+    // TODO(njriasan): This only works because
+    // producer->getResult(0) already has a ttg.slice attribute.
+    // We will need to update this to work more generally.
```

Comment:
This is a bug I hit writing a test. Basically I need to add the logic for `ttg.slice`. This only worked because its behind an expandDims. Since that is the example I will do this in a followup later today since it should be non-blocking.